### PR TITLE
Fix statistics header file inclusion [11614]

### PR DESCRIPTION
**Pull request by richiware**
Depending on the order of header file inclusion could occur a linkage error. This is because `StatisticsBase.hpp` always includes `StatiticsCommon.hpp`, with statistics enable or not. This PR moves content of `StatisticsCommonEmpty.hpp` into `StatisticsCommon.hpp` and uses the `FASTDDS_STATISTICS` define to select implementation.

### CODE REVIEW

**Review by MiguelBarro — CHANGES_REQUESTED** [review]
LGTM and to Github actions
Please note the comment

File: include/fastdds/statistics/rtps/StatisticsCommon.hpp [hunk 1]
```diff
@@ -256,6 +258,123 @@ class StatisticsReaderImpl
             uint32_t payload);
 };
 
+#else
```
[the remark this answers is not in the document]
Comment (richiware) [reply include/fastdds/statistics/rtps/StatisticsCommon.hpp]: done

**Review by MiguelBarro — APPROVED** [review]
LGTM

**Comment by richiware** [comment]
Failed tests not related. Merging...

**Pull request merged**